Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/SelectionComponent.cpp

```diff
@@ -5,90 +5,84 @@
-        }
+        static constexpr const char* const SelectionComponentRemovalNotice =
+            "[INFORMATION] The editor SelectionComponent is being removed from .prefab data. Please edit and save the level to persist "
+            "the change.";
 
-        SelectionComponent::~SelectionComponent()
+        class SelectionComponentSerializer : public AZ::BaseJsonSerializer
         {
-        }
+        public:
+            AZ_RTTI(SelectionComponentSerializer, "{469D6DA3-0960-4A84-B806-4FCD0C7A7EBC}", AZ::BaseJsonSerializer);
+            AZ_CLASS_ALLOCATOR_DECL;
 
-        void SelectionComponent::Init()
-        {
-        }
+            SelectionComponentSerializer() = default;
+            ~SelectionComponentSerializer() override = default;
 
-        void SelectionComponent::UpdateBounds(const AZ::Aabb& newBounds)
-        {
-            m_currentSelectionAABB = newBounds;
-        }
+            AZ::JsonSerializationResult::Result Load(
+                void* outputValue,
+                const AZ::Uuid& outputValueTypeId,
+                const rapidjson::Value& inputValue,
+                AZ::JsonDeserializerContext& context) override;
 
-        void SelectionComponent::Activate()
-        {
-            SelectionComponentMessages::Bus::Handler::BusConnect(GetEntityId());
-            AzToolsFramework::EntitySelectionEvents::Bus::Handler::BusConnect(GetEntityId());
-        }
+            AZ::JsonSerializationResult::Result Store(
+                rapidjson::Value& outputValue,
+                const void* inputValue,
+                const void* defaultValue,
+                const AZ::Uuid& valueTypeId,
+                AZ::JsonSerializerContext& context) override;
+        };
 
-        void SelectionComponent::Deactivate()
-        {
-            AzToolsFramework::EntitySelectionEvents::Bus::Handler::BusDisconnect(GetEntityId());
-            SelectionComponentMessages::Bus::Handler::BusDisconnect(GetEntityId());
+        AZ_CLASS_ALLOCATOR_IMPL(SelectionComponentSerializer, AZ::SystemAllocator, 0);
 
-            m_currentSelectionFlag = SF_Unselected;
-        }
-
-        void SelectionComponent::OnSelected()
-        {
-            m_currentSelectionFlag = SF_Selected;
-        }
-
-        void SelectionComponent::OnDeselected()
+        AZ::JsonSerializationResult::Result SelectionComponentSerializer::Load(
+            [[maybe_unused]] void* outputValue,
+            [[maybe_unused]] const AZ::Uuid& outputValueTypeId,
+            [[maybe_unused]] const rapidjson::Value& inputValue,
+            AZ::JsonDeserializerContext& context)
         {
-            m_currentSelectionFlag = SF_Unselected;
+            namespace JSR = AZ::JsonSerializationResult;
+            return context.Report(JSR::Tasks::ReadField, JSR::Outcomes::Unavailable, SelectionComponentRemovalNotice);
         }
 
-        bool SelectionComponent::IsSelected() const
+        AZ::JsonSerializationResult::Result SelectionComponentSerializer::Store(
```

Comment:
If it is not possible to create the selection component and it can't be loaded from previous files, do we need the store function at all? If there is no selection component object created in memory, then there won't be need to store it right?

Reply:
Hmm that's true 🤔 I potentially could remove this... I'd thought keeping it here to explicitly not handle it is the safer option, but you might be right that the default op in BaseJsonSerializer is sufficient (see below)

```
JsonSerializationResult::Result BaseJsonSerializer::Store(rapidjson::Value& outputValue, const void* inputValue, const void* defaultValue, const Uuid& valueTypeId, JsonSerializerContext& context)
{
    JsonSerializationResult::ResultCode result(JsonSerializationResult::Tasks::WriteValue);
    result.Combine(ContinueStoring(outputValue, inputValue, defaultValue, valueTypeId, context, ContinuationFlags::IgnoreTypeSerializer));
    return context.Report(result, "Ignoring custom serialization during store");
}
```
I'll give it a quick test and see :+1: